src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrder.java

```diff
@@ -0,0 +1,165 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.TextBlock;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+
+ */
+@StatelessCheck
+public class JavadocParamOrder extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    @Override
+    public final int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.RECORD_DEF,
+        };
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.ANNOTATION_FIELD_DEF,
+            TokenTypes.RECORD_DEF,
+            TokenTypes.COMPACT_CTOR_DEF,
+        };
```

Comment:
there are things i would like to be checked, please check this i was not sure about this.

Reply:
Not sure, please check if it can be applied also to [ENUM_CONSTANT_DEF](https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF), [RECORD_COMPONENT_DEF](https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_COMPONENT_DEF).
 @pbludov Sir, please confirm these tokens.

Reply:
No.
`ENUM_CONSTANT_DEF` can not be annotated with `@param` tag, it should be at the enum `ctor`.
Javadoc for `RECORD_COMPONENT_DEF` is not allowed, they should be documented at `RECORDT_DEF` level.

For the other tags:
* `ENUM_DEF` NOT OK. There is no generic `enum` in Java. At least, until http://openjdk.java.net/jeps/301
* `COMPACT_CTOR_DEF` NOT OK. It has no explicit parameters, so they can not be annotated.

Reply:
* `ANNOTATION_FIELD_DEF` NOT OK, annotation fields can not have parameters nor generic type arguments.

Reply:
`CTOR_DEF` should probably be added.

Reply:
CTOR_DEF in getAcceptableTokens ?